django/db/backends/base/references.py

```diff
@@ -0,0 +1,129 @@
+"""
+Helpers to manipulate DDL statements that might need to be adjusted or
+discarded within a schema edition lifecycle.
+"""
+
+
+class Reference:
+    """Base class that defines the reference interface."""
+
+    def references_table(self, table):
+        """
+        Return whether or not this instance references the specified table.
+        """
+        return False
+
+    def references_column(self, table, column):
+        """
+        Return whether or not this instance references the specified column.
+        """
+        return False
+
+    def __repr__(self):
+        return '<%s %r>' % (self.__class__.__name__, str(self))
+
+    def __str__(self):
+        """Subclasses must define how they should be converted to string."""
+        raise NotImplementedError
```

Comment:
Perhaps this exception should have a message rather than a docstring (see existing uses of NotImplementedError)